docs/source/guides/workers.rst

```diff
@@ -0,0 +1,81 @@
+=======
+Workers
+=======
+
+BentoML workers enhance the parallel processing capabilities of machine learning models. Under the hood, there are one or multiple workers within a BentoML :doc:`Service </guides/services>`. They are the processes that actually run the code logic within the Service. This design leverages the parallelism of the underlying hardware, whether it's multi-core CPUs or multi-device GPUs.
+
+This document explains how to configure and allocate workers for different use cases.
+
+Configure workers
+-----------------
+
+When you define a BentoML Service, use the ``workers`` parameter to set the number of workers. For example, setting ``workers=4`` launches four worker instances of the Service, each running in its process. Each worker is homogeneous, which means they perform the same tasks.
+
+.. code-block:: python
+
+    @bentoml.service(
+        workers=4,
+    )
+    class MyService:
+        # Service implementation
+
+The number of workers isn't necessarily equivalent to the number of concurrent requests a BentoML Service can serve in parallel. With optimizations like :doc:`adaptable batching </guides/adaptive-batching>` and continuous batching, each worker can potentially handle many requests simultaneously to enhance the throughput of your Service. To specify the ideal number of concurrent requests for a Service (namely, all workers within the Service), you can configure :doc:`concurrency </guides/concurrency>`.
+
+Use cases
+---------
+
+Workers allow a BentoML Service to effectively utilize underlying hardware accelerators, like CPUs and GPUs, ensuring optimal performance and resource utilization.
+
+The default worker count in BentoML is set to ``1``. However, depending on your computational workload and hardware configuration, you might need to adjust this number.
+
+CPU workloads
+^^^^^^^^^^^^^
+
+Python processes are subject to the Global Interpreter Lock (GIL), which limits the execution of multiple threads in a single process. To avoid this and fully leverage multi-core CPUs, you can start multiple workers. However, be mindful of the memory implications, as each worker will load a copy of the model into memory. Ensure that your machine's memory can support the cumulative memory requirements of all workers.
```

Comment:
```
 be mindful of the memory implications, as each worker will load a copy of the model into memory
```
Not sure about this.

Reply:
I remember this is the true - Each work will load the model into memory, but feel free to correct me if I am wrong. @bojiang @ssheng

Reply:
Add some explanations. Hopefully they can be helpful here.